### DISCUSSION TITLE
Support fisheye cameras

### DISCUSSION THREAD

**Discussion by charlielito** [foxglove/community#901] [Q&A]
I would be very nice if studio could support fisheye distortion model cameras. I cannot use it right now

![image (82)](https://github.com/foxglove/community/assets/8033598/0c14f349-82a2-4c22-8cdc-da7c3760438e)

**Comment by banisadr** [comment]
@charlielito is there a specific fisheye distortion model you're looking for? You might be able to leverage the existing [rational polynomial](https://docs.foxglove.dev/docs/visualization/message-schemas/camera-calibration/#panel-support) model which can be used to model fisheye lenses of moderate distortion. If you have a high FOV, the existing pin-hole camera model used may not work well regardless of the distortion model.

**Comment by snosenzo** [comment]
@charlielito the only strings that we support for the `model` field are `plumb_bob` and `rational_polynomial`. Both I believe are functionally the same in our rendering code. As @banisadr said you should be able to use `rational_polynomial` with the right distortion parameters. However if that doesn't work we'd be happy to help add support for a higher FOV camera model if you provide some example data with expected outputs.

**Reply by charlielito** [reply]
We do have high FOV cameras, with high radial distortion, that's why when we calibrate them we use the fisheye distortion model from opencv https://docs.opencv.org/4.x/db/d58/group__calib3d__fisheye.html

What would you need for this? An example calibration file and image and the code to reproduce the undistortion with this model?
